src/Analysis/SnippetAnalysis.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @license
+ *
+ * Copyright (C) debricked AB
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code (usually found in the root of this application).
+ */
+
+namespace App\Analysis;
+
+class SnippetAnalysis
+{
+    /** @var string[] */
+    private array $ignore;
+
+    /** @var string[] */
+    private array $snippets;
+
+    public function __construct()
+    {
+        $this->ignore = ['bmp' => '', 'class' => '', 'conf' => '', 'csv' => '', 'eps' => '', 'gif' => '',
+            'gitignore' => '', 'gz' => '', 'jpeg' => '', 'jpg' => '', 'lock' => '', 'md' => '', 'mp3' => '',
+            'mp4' => '', 'pdf' => '', 'png' => '', 'rst' => '', 'sql' => '', 'tif' => '', 'zip' => '', ];
+        $this->snippets = [];
+    }
+
+    /**
+     * Analyses a file, calculates its hashes (if it is a relevant file), and adds the hashes to the internal state.
+     *
+     * @param string $absoluteFilename the filename to analyse
+     * @param string $relativeFilename the filename to include in the WFP fingerprint
+     *
+     * @return bool true if the file was analysed, false if the file was ignored
+     */
+    public function analyseFile(string $absoluteFilename, string $relativeFilename): bool
+    {
+        // First determine if we want to scan it or not by checking the ignore list.
+        $extension = \pathinfo($absoluteFilename, PATHINFO_EXTENSION);
+        if (empty($extension) || \array_key_exists(\strtolower($extension), $this->ignore)) {
```

Comment:
Do we want this to analyse e.g. WORKSPACE files? They have no extension

Reply:
I would say that we don't care about them from a license perspective. I don't think they contain much code anyway. All files without extensions are also ignored in the upstream SCANOSS example integrations, so they don't seem to expect them to be scanned.

However, I'm open for discussions/suggestions regarding the list of ignored extensions, if you think anything needs to be changed.

Reply:
Added a few image formats :)